### PR TITLE
overmap borders are more seamless for AI, and for shooting

### PR DESCRIPTION
**Pull request by Ikalpo**
## About The Pull Request

Adds `overmap_angle()` and `overmap_dist()`, new versions of `get_Angle()` and `get_Dist()` that take into account how the edges of the overmap loop.
In the case of `overmap_dist()`, it is also more accurate than `get_Dist()`.
[Here's a little demonstration of how it looks in Desmos:](https://www.desmos.com/calculator/6akddpjzio)

This results in the AI being smarter about the edges of the overmap, being able to take a more efficient route to its destination, and to hit targets on the other side of the border.

This also fixes where you would click on or past the mirage border, and it would shoot in the opposite direction.

And also, make the radar range a circle instead of a square.

## Why It's Good For The Game

kills border cheese dead, and also it's a bit of a bugfix

## Changelog
:cl:
balance: the AI is smarter with the edges of the overmap, no more hiding in corners!
fix: fixed firing near the edge of the overmap would shoot in the wrong direction
tweak: radar range is no longer a square
/:cl:

### CODE REVIEW

**Comment by Ikalpo** [comment]
> This pull request has conflicts, please resolve those before we can evaluate the pull request.

fug

**Review by TheFakeElon — COMMENTED** [review]
If you're going to do this you should probably make the radar seamless too, otherwise you'll have blips on the other side of the map shooting you from behind

**Comment by Ikalpo** [comment]
> If you're going to do this you should probably make the radar seamless too, otherwise you'll have blips on the other side of the map shooting you from behind

yeah I had a look at radar code.
hell

> This will almost always be guaranteed to be true, if it's not, then it's a problem with the calling proc and should be fixed

blame Get_Angle that's where I got that from

> magic numbers bad

yeah but I tried to find the scientific number for map size and couldn't find it (127.5 is 255/2)
if someone can find that for me, that'd be appreciated

**Comment by TheFakeElon** [comment]
> yeah but I tried to find the scientific number for map size and couldn't find it (127.5 is 255/2)
if someone can find that for me, that'd be appreciated

`world.maxx` & `world.maxy`

**Comment by red031000** [comment]
borders are not looping for the main ship, at all, seems AI also has troubles with this

**Comment by Ikalpo** [comment]
> borders are not looping for the main ship, at all, seems AI also has troubles with this

wasn't me #1870

File: nsv13/code/__HELPERS/overmap.dm [hunk 1]
```diff
@@ -60,3 +60,58 @@ Helper method to get what ship an observer belongs to for stuff like parallax.
...
+	return sqrt(CX**2 + CY**2)
+
+/**
+Another get_angle that works better with the looping edges of the overmap
+*/
+
+/proc/overmap_angle(atom/A,atom/B)
+	if (!A || !B)
+		return 0
+	var/T = 127.5 - (TRANSITIONEDGE + 1)
```
[the remark this answers is not in the document]
Comment (Bokkiewokkie) [reply nsv13/code/__HELPERS/overmap.dm:94]: @Ikalpo Can you please add world.maxx and world.maxy in here then?

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**